node-api/handlers/proof/merkle/validator_balance_test.go

```diff
@@ -0,0 +1,265 @@
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package merkle_test
+
+import (
+	"testing"
+
+	ctypes "github.com/berachain/beacon-kit/consensus-types/types"
+	"github.com/berachain/beacon-kit/node-api/handlers/proof/merkle"
+	"github.com/berachain/beacon-kit/node-api/handlers/proof/merkle/mock"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/math"
+	"github.com/berachain/beacon-kit/primitives/version"
+	"github.com/stretchr/testify/require"
+)
+
+// TestValidatorBalanceProof_Comparison tests the balance function
+// and that the generated proof matches the expected proof.
+func TestValidatorBalanceProof_Comparison(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name              string
+		forkVersion       common.Version
+		numValidators     int
+		slot              math.Slot
+		valIndex          math.ValidatorIndex
+		parentBlockRoot   common.Root
+		bodyRoot          common.Root
+		balance           uint64
+		expectedProofFile string
+	}{
+		{
+			name:              "1 Validator Set",
+			forkVersion:       version.Electra(),
+			numValidators:     1,
+			slot:              4,
+			valIndex:          0,
+			parentBlockRoot:   common.Root{1, 2, 3},
+			bodyRoot:          common.Root{3, 2, 1},
+			balance:           1000000000000000000,
+			expectedProofFile: "one_validator_validator_balance_proof.json",
+		},
+		{
+			name:              "Many Validator Set",
+			forkVersion:       version.Electra(),
+			numValidators:     100,
+			slot:              5,
+			valIndex:          95,
+			parentBlockRoot:   common.Root{1, 2, 3, 4, 5, 6},
+			bodyRoot:          common.Root{3, 2, 1, 9, 8, 7},
+			balance:           2000000000000000000,
+			expectedProofFile: "many_validators_validator_balance_proof.json",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			vals := make(ctypes.Validators, tc.numValidators)
+			for i := range vals {
+				vals[i] = &ctypes.Validator{}
+			}
+
+			bs := mock.NewBeaconStateWith(
+				tc.slot, vals, 0, common.ExecutionAddress{}, tc.forkVersion,
+			)
+
+			bs.Balances = make([]uint64, tc.numValidators)
+			bs.Balances[tc.valIndex] = tc.balance
+
+			bbh := ctypes.NewBeaconBlockHeader(
+				tc.slot,
+				tc.valIndex,
+				tc.parentBlockRoot,
+				bs.HashTreeRoot(),
+				tc.bodyRoot,
+			)
+
+			proof, _, _, err := merkle.ProveBalanceInBlock(tc.valIndex, bbh, bs, bs.Balances)
+			require.NoError(t, err)
+			expectedProof := ReadProofFromFile(t, tc.expectedProofFile)
```

Comment:
@bearmigiano you can use this code to re-create the balance proofs for specific data as needed in contracts testing